Dockerfile

```diff
@@ -237,16 +237,16 @@ RUN apk add --no-cache shadow shadow-uidmap sudo vim iptables fuse \
 ENV BUILDKIT_INTEGRATION_CONTAINERD_EXTRA="containerd-1.4=/opt/containerd-alt-14/bin,containerd-1.5=/opt/containerd-alt-15/bin"
 ENV BUILDKIT_INTEGRATION_SNAPSHOTTER=stargz
 ENV CGO_ENABLED=0
-COPY --from=stargz-snapshotter /out/* /usr/bin/
-COPY --from=rootlesskit /rootlesskit /usr/bin/
-COPY --from=containerd-alt-14 /out/containerd* /opt/containerd-alt-14/bin/
-COPY --from=containerd-alt-15 /out/containerd* /opt/containerd-alt-15/bin/
-COPY --from=registry /bin/registry /usr/bin
-COPY --from=runc /usr/bin/runc /usr/bin
-COPY --from=containerd /out/containerd* /usr/bin/
-COPY --from=cni-plugins /opt/cni/bin/bridge /opt/cni/bin/host-local /opt/cni/bin/loopback /opt/cni/bin/
-COPY hack/fixtures/cni.json /etc/buildkit/cni.json
-COPY --from=binaries / /usr/bin/
+COPY --link --from=stargz-snapshotter /out/* /usr/bin/
+COPY --link --from=rootlesskit /rootlesskit /usr/bin/
+COPY --link --from=containerd-alt-14 /out/containerd* /opt/containerd-alt-14/bin/
+COPY --link --from=containerd-alt-15 /out/containerd* /opt/containerd-alt-15/bin/
+COPY --link --from=registry /bin/registry /usr/bin/
+COPY --link --from=runc /usr/bin/runc /usr/bin/
+COPY --link --from=containerd /out/containerd* /usr/bin/
+COPY --link --from=cni-plugins /opt/cni/bin/bridge /opt/cni/bin/host-local /opt/cni/bin/loopback /opt/cni/bin/
+COPY --link hack/fixtures/cni.json /etc/buildkit/cni.json
```

Comment:
Do we want to use `--link` with the local ref here? Obviously that works but currently results in finalizing the local ref making it unavailable for re-use.

Reply:
Why would it finalize? The local ref itself is not the merge input.